graphs/util/db.py

```diff
@@ -4742,3 +4744,112 @@ def constructResponse(statusCode, message, error):
+		@param group_id: id of the grop
+		@param member_id: id of the member the graph is shared
+	'''
+	# Create database connection
+	db_session = data_connection.new_session()
+
+	# Get the current time
+	cur_time = datetime.now()
+
+	new_event = models.ShareGraphEvent(graph_id=graph_id, owner_id=owner_id, group_id=group_id, member_id=member_id, share_time=cur_time, event_active=True)
+	db_session.add(new_event)
+	db_session.commit()
+	db_session.close()
+
+
+def update_share_graph_event(event_id, active, member_id):
+	'''
+		Update the share graph event. Change its active state.
+		If active is True then the notification is not read/clicked.
+
+		@param event_id: id of the share graph event
+		@param active: Boolean value, update the state of event
+		@param member_id: id of the user, the logged in user.
+	'''
+	db_session = data_connection.new_session()
+
+	try:
+		event = db_session.query(models.ShareGraphEvent).filter(models.ShareGraphEvent.id == event_id).filter(models.ShareGraphEvent.member_id == member_id).one()
+		event.event_active = active
+		db_session.commit()
+		db_session.close()
+	except NoResultFound:
+		db_session.close()
+		return 'Event not found'
+
+
+def delete_share_graph_event(event_id, member_id):
+	'''
+		Delete the share graph event from the table for the member
+
+		@param event_id: id of the share graph event
+		@param member_id: id of the member
+
+	'''
+	db_session = data_connection.new_session()
+	try:
+		event = db_session.query(models.ShareGraphEvent).filter(models.ShareGraphEvent.id == event_id).filter(models.ShareGraphEvent.member_id == member_id).one()
+		db_session.delete(event)
+		db_session.commit()
+		db_session.close()
+	except NoResultFound:
+		db_session.close()
+		return 'Event not found'
+
+
+def get_share_graph_event_by_member_id(member_id):
+	'''
+		Return all the share graph events for a user
+
+		@param member_id: id of the user
+	'''
+	# Create database connection
+	db_session = data_connection.new_session()
+
+	try:
+		events = db_session.query(models.ShareGraphEvent).filter(models.ShareGraphEvent.member_id == member_id).all()
+		db_session.close()
+		return events
+	except NoResultFound:
+		db_session.close()
+		return None
+
+
+def get_share_graph_event_by_id(event_id, member_id):
+	'''
+		Return share graph event notification
+
+		@param event_id: id of the event
+		@param member_id: id of the logged in user
+	'''
+	db_session = data_connection.new_session()
+	try:
+		event = db_session.query(models.ShareGraphEvent).filter(models.ShareGraphEvent.id == event_id).filter(models.ShareGraphEvent.member_id == member_id).one()
+		db_session.close()
+		return event
+	except NoResultFound:
+		db_session.close()
+		return 'Event not found'
+
+
+def get_all_share_graph_event():
+	'''
+		Return all the share graph events.
+	'''
+	db_session = data_connection.new_session()
+	events = db_session.query(models.ShareGraphEvent).all()
+	db_session.close()
+	return events
+
```

Comment:
Add another method called `set_share_graph_events_inactive(event_ids)`.

> events: list of event_ids

This function should set all events as inactive. Return type should be None. Raise an error if query fails.

Try to make it a single db query. You can use the update api in sqlalchemy.

>